R/zchunk_LB165.ag_water_R_C_Y_GLU_irr.R

```diff
@@ -0,0 +1,407 @@
+    # gridded and national inventories. These decisions are documented below.
+
+    # The inventory data disaggregate green, blue, and gray water. Gray water indicates
+    # downstream pollution; this is not considered in GCAM and has not been included in the
+    # gridded data processing. Blue water is assigned to only irrigated production, but green
+    # water applies to both irrigated and rainfed production, as the two management technologies
+    # are not disaggregated in the inventory data. For rainfed crops, total biophysical = green,
+    # and for irrigated crops, total biophysical = blue + green.
+
+    # Initial data cleaning (original lines 55-57)
+    L100.Water_footprint_m3 %>%
+      rename(MH_crop = GTAP_crop) %>%
+      left_join_error_no_match(select(FAO_ag_items_PRODSTAT, MH_crop, GTAP_crop), by = "MH_crop") ->
+      L165.Water_footprint_m3
+
+    # Re-cast nation-level data so that blue and green are represented as data columns (original lines 59-64)
+    Mekonnen_Hoekstra_Rep47_A2 %>%
+      gather(iso, coef_m3t, -FAO_crop, -water_type) %>%
+      mutate(coef_m3kg = coef_m3t / CONV_T_KG) %>%
+      select(iso, FAO_crop, water_type, coef_m3kg) %>%
+      # Set all missing values to zero -- this may be re-visited at another point
+      # We may want to substitute default values from the MH2011 table
+      mutate(coef_m3kg = if_else(is.na(coef_m3kg), 0, coef_m3kg)) %>%
+      spread(water_type, coef_m3kg) ->
+      L165.Mekonnen_Hoekstra_Rep47_A2
+
+    # Build inventory of water use by country, crop, and GLU for the 18 gridded crops (lines 66-74 in original)
+    # We give precedence to the aggregated gridded data for the 18 crops that are mapped,
+    # and fill in nation-level inventory data for the remainder
+    L165.ag_Prod_t_ctry_crop_GLU <- rename(L100.LDS_ag_prod_t, Prod_t = value)
+
+    L165.Water_footprint_m3 %>%
+      filter(water_type %in% c("green", "blue")) %>%
+      # TODO: this following join has NAs but shouldn't change # of rows (currently doesn't) - use left_join_restricted
+      left_join(L165.ag_Prod_t_ctry_crop_GLU, by = c("iso", "GTAP_crop", "GLU")) %>%
+      # M+H have some country/crop/GLU combinations that Monfreda doesn't
+      # Set missing values to 0
+      mutate(Prod_t = if_else(is.na(Prod_t), 0L, Prod_t)) %>%
+      # For the 18 M+H gridded crops, calculate the coefs by country, GLU, and crop (original lines 79-82)
+      # NOTE: dropping all country / GLU / crops with zero production; these won't have any water anyway
+      filter(Prod_t > 0) %>%
+      mutate(coef_m3kg = value / (Prod_t * CONV_T_KG)) ->
+      L165.ag_Water_ctry_MHcrop_GLU
+
+    # Clip extremely high values which are due to statistical discrepancies (original lines 84-106)
+    # The assumed maximum values are crop specific and based on analysis of the 2011 inventory data
+    # In this step, we're adding 2 m3/kg to every maximum observed value across all national averages.
```

Comment:
Can we rephrase this slightly? When I first read it I thought you were adding 2 to the data, not setting the cap at max + 2.